src/daft-core/src/array/ops/list.rs

```diff
@@ -112,6 +112,55 @@ impl ListArray {
             Box::new(arrow2::array::Utf8Array::from_iter(result)),
         )))
     }
+
+    fn get_children_helper(
+        &self,
+        idx_iter: &mut impl Iterator<Item = Option<i64>>,
+    ) -> DaftResult<Series> {
+        let mut growable = make_growable(
+            self.name(),
+            self.child_data_type(),
+            vec![&self.flat_child],
+            true,
+            self.len(),
+        );
+
+        let offsets = self.offsets();
+
+        for i in 0..self.len() {
+            let is_valid = self.is_valid(i);
+            let start = *offsets.get(i).unwrap();
+            let end = *offsets.get(i + 1).unwrap();
+            let child_idx = idx_iter.next().unwrap().unwrap();
+
+            // only add index value when list is valid and index is within bounds
+            match (is_valid, child_idx >= 0, start + child_idx, end + child_idx) {
```

Comment:
I think matches start to get unwieldy once you start hitting 3/4 things to match on, we could simplify it here:
```
if is_valid {
    let idx_offset = if child_idx >= 0 {start + child_idx} else {end + child_idx};
    growable.extend(0, idx_offset as usize, 1);
} else {
    growable.add_nulls();
}
```